mojo/stdlib/stdlib/gpu/host/dim.mojo

```diff
@@ -193,3 +193,36 @@ struct Dim(Stringable, Writable):
             The value of the x dimension.
         """
         return self[0]
+
+    fn set_z[T: Indexer](mut self, z: T):
```

Comment:
what do you think about using setattr and getattr here (similar to what is done for ThreadIdx struct)?